evm_loader/program/src/evm/buffer.rs

```diff
@@ -221,17 +209,116 @@ impl<'de> serde::Deserialize<'de> for Buffer {
 
                 let (index, variant) = data.variant::<u32>()?;
                 match index {
-                    0 => variant.unit_variant().map(|_| Buffer::empty()),
                     1 => variant.newtype_variant().map(Buffer::from_slice),
                     2 => variant.struct_variant(&["key", "range"], self),
                     _ => Err(serde::de::Error::unknown_variant(
                         "_",
-                        &["empty", "owned", "account"],
+                        &["owned", "account"],
                     )),
                 }
             }
         }
 
-        deserializer.deserialize_enum("evm_buffer", &["empty", "owned", "account"], BufferVisitor)
+        deserializer.deserialize_enum("evm_buffer", &["owned", "account"], BufferVisitor)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::cell::RefCell;
+    use std::rc::Rc;
+
+    macro_rules! assert_slice_ptr_eq {
+        ($actual:expr, $expected:expr) => {{
+            let actual: &[_] = $actual;
+            let (expected_ptr, expected_len): (*const _, usize) = $expected;
+            assert_eq!(actual.as_ptr(), expected_ptr);
+            assert_eq!(actual.len(), expected_len);
+        }};
+    }
+
+    #[test]
+    fn test_deref_owned_empty() {
+        let data = Vec::default();
+        let expected = (data.as_ptr(), data.len());
+        assert_slice_ptr_eq!(&*Buffer::default(), expected);
+    }
+
+    #[test]
+    fn test_deref_owned_non_empty() {
+        let data = vec![1];
+        let expected = (data.as_ptr(), data.len());
+        assert_slice_ptr_eq!(&*Buffer::from_vec(data), expected);
+    }
+
+    struct OwnedAccountInfo {
+        key: Pubkey,
+        lamports: u64,
+        data: Vec<u8>,
+        owner: Pubkey,
+        rent_epoch: u64,
+        is_signer: bool,
+        is_writable: bool,
+        executable: bool,
+    }
```

Comment:
There is already a `crate::executor::OwnedAccountInfo` type which can be imported here.

Reply:
Looks like we should do a follow up PR to clean this up.

Reply:
See https://github.com/neonlabsorg/neon-evm/pull/260.